snapcraft/internal/repo/_base.py

```diff
@@ -270,19 +270,42 @@ def _fix_xml_tools(cls, unpackdir: str) -> None:
             )
 
     @classmethod
-    def _fix_symlink(cls, path: str, unpackdir: str, root: str) -> None:
-        host_target = os.readlink(path)
-        if host_target in cls.get_package_libraries("libc6"):
-            logger.debug(
-                "Not fixing symlink {!r}: it's pointing to libc".format(host_target)
-            )
-            return
+    def _fix_absolute_symlink(cls, path: str, unpackdir: str) -> None:
+        """Fix absolute symlinks that would break in snap.
+
+        If it points to something in the libc package, just ignore it.
 
-        target = os.path.join(unpackdir, os.readlink(path)[1:])
-        if not os.path.exists(target) and not _try_copy_local(path, target):
+        The path may point to something that exists relative to unpackdir.
+        If it exists, update symlink to point to that.
+
+        Next check to see if symlink points to a valid file on the host.
+        If it does, copy it from host and warn.
+
+        Otherwise, warn about dangling symlink.
+        """
+        target = os.readlink(path)
+        if not os.path.isabs(target):
             return
-        os.remove(path)
-        os.symlink(os.path.relpath(target, root), path)
+
+        real_path = os.path.realpath(path)
+        unpack_target = os.path.join(unpackdir, target[1:])
+
+        if target in cls.get_package_libraries("libc6"):
+            # Ignore libc links.
+            logger.debug(
+                "Not fixing symlink {!r}: it's pointing to libc".format(target)
+            )
+        elif os.path.exists(unpack_target):
+            # Update link to be relative to unpackdir, if exists.
+            os.remove(path)
+            os.symlink(os.path.relpath(unpack_target, os.path.dirname(path)), path)
+        elif os.path.isfile(real_path):
+            # Copy file from host, if exists.
+            logger.warning("Copying needed target from the system {}".format(real_path))
+            os.makedirs(os.path.dirname(target), exist_ok=True)
+            shutil.copyfile(real_path, target)
```

Comment:
target here is the read link, and real_path is the resolved link which may lead to the same file.
```suggestion
            shutil.copyfile(real_path, path)
```

Although I think we need better variable names here to better represent what is going on